### PR TITLE
Generalised better initialisation.

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

**Comment by benleetownsend** [comment]
Pending final experiments for parity with Madison/fancy_init branch on classification.

File: finetune/base.py [hunk 1]
```diff
@@ -177,6 +177,24 @@ def finetune(self, Xs, Y=None, batch_size=None):
 
         with warnings.catch_warnings():
             warnings.simplefilter("ignore")
+            if self.config.prefit_init:
+                tf.logging.info("Starting pre-fit initialisation...")
+                num_layers_trained = self.config.num_layers_trained
+                self.config.num_layers_trained = 0
+                estimator.train(train_input_fn, hooks=train_hooks, steps=num_steps)
```
Comment (madisonmay) [inline finetune/base.py]: There's some room here for debating whether or not train_hooks / num_steps should be shared or whether this portion should always work the same way.

File: finetune/base.py [hunk 2]
```diff
@@ -177,6 +177,24 @@ def finetune(self, Xs, Y=None, batch_size=None):
...
+                self.config.num_layers_trained = num_layers_trained
+                self.saver.variables = {k: v for k, v in self.saver.variables.items() if "adam" not in k and "global_step" not in k}
+                for weight in self.saver.variables:
+                    if weight.startswith("model/target/"):
+                        w = self.saver.variables[weight]
+                        shape = w.shape
```
Comment (madisonmay) [inline finetune/base.py]: Feels weird to alias this to save two characters but whatevs.  Some weird spacing w/ len(shape) below.

File: finetune/base.py (hunk 1 above)
Comment (madisonmay) [reply finetune/base.py]: For now we're going to leave this as is.

**Pull request merged**